extensions/ql-vscode/test/vscode-tests/no-workspace/model-editor/suggestion-queries.test.ts

```diff
@@ -0,0 +1,88 @@
+import { createMockLogger } from "../../../__mocks__/loggerMock";
+import type { DatabaseItem } from "../../../../src/databases/local-databases";
+import { DatabaseKind } from "../../../../src/databases/local-databases";
+import { file } from "tmp-promise";
+import { QueryResultType } from "../../../../src/query-server/messages";
+import { QueryLanguage } from "../../../../src/common/query-language";
+import { mockedObject, mockedUri } from "../../utils/mocking.helpers";
+import { Mode } from "../../../../src/model-editor/shared/mode";
+import { join } from "path";
+import type { CodeQLCliServer } from "../../../../src/codeql-cli/cli";
+import type { QueryRunner } from "../../../../src/query-server";
+import { QueryOutputDir } from "../../../../src/local-queries/query-output-dir";
+import { runSuggestionsQuery } from "../../../../src/model-editor/suggestion-queries";
+
+describe("runSuggestionsQuery", () => {
+  it("should run query", async () => {
+    const language = QueryLanguage.Ruby;
+    const outputDir = new QueryOutputDir(join((await file()).path, "1"));
+
+    const options = {
+      parseResults: jest.fn().mockResolvedValue([]),
```

Comment:
Can we test that this function is called twice with mock results from `bqrsDecodeAll` and that the function returns the first and second results inside an object with `input` and `output` keys? I think that would add some value to ensure that the flow is correct.